### ISSUE TITLE
Windows integrated authentification - useAppPoolCredentials=true

### ISSUE THREAD

**Issue by matt6697** [PowerShell/xWebAdministration#322]
Kerberos authentification on a web application running under a service account (Application Pool Identity / SpecificUser) requires to setup the following variables under system.webServer/security/authentication/windowsAuthentication

useAppPoolCredentials = true
useKernelMode=false

xWebAppPool allowed me to configure the Application Pool Identity service account on my application pool but how can I modify the useAppPoolCredentials and useKernelMode using this DSC module as the xWebConfigKeyValue seems to be restricted to AppSettings ?

**Comment by remcoeissing** [comment]
That can currently not be done with a builtin DSC Resource. You could use the Script resource to achieve this.

`
Configuration WindowsAuthenticationExample
{
    param(
        [Parameter(Mandatory = $true)]
        [String]
        $Site,

        [Parameter(Mandatory = $true)]
        [Boolean]
        $AppPoolValue
    )

    Import-DscResource –ModuleName 'PSDesiredStateConfiguration'

    Script UseAppPoolCredentials
    {
        GetScript = { 
            $useAppPoolCredentials = Get-WebConfigurationProperty -pspath 'MACHINE/WEBROOT/APPHOST' -location $using:site -filter 'system.webServer/security/authentication/windowsAuthentication' -name 'useAppPoolCredentials'
            return @{ Result = $useAppPoolCredentials.Value }
        }          
        TestScript = { 
            $useAppPoolCredentials = Get-WebConfigurationProperty -pspath 'MACHINE/WEBROOT/APPHOST' -location $using:site -filter 'system.webServer/security/authentication/windowsAuthentication' -name 'useAppPoolCredentials'
            return ($AppPoolValue -eq $useAppPoolCredentials.Value)
        }
        SetScript = { 
            Set-WebConfigurationProperty -pspath 'MACHINE/WEBROOT/APPHOST' -location $using:site -filter 'system.webServer/security/authentication/windowsAuthentication' -name 'useAppPoolCredentials' -value $using:AppPoolValue
        }
    }
}`

You only require useAppPoolCredentials set to true. Kernel mode should stay enabled and you need to register the SPN to the SpecificUser.

**Comment by matt6697** [comment]
Works like a charm ;-) Tks

**Comment by matt6697** [comment]
Would be great to integrate this as a DSC resource. Any plan to add this to a next sprint ?

**Comment by johlju** [comment]
I label this as help wanted so that someone in the community can run with this.

**Comment by matt6697** [comment]
Will this configuration new be possible through xWebConfigProperty / xWebConfigPropertyCollection ?

**Comment by regedit32** [comment]
@matt6697 Yes, xWebConfigProperty would be the right resource for this.  Here's an example how to set the useAppPoolCredentials property:
```
Configuration WindowsAuthenticationExample
{
    Import-DscResource -ModuleName xWebAdministration

    xWebConfigProperty useAppPoolCredentials
    {
            WebsitePath  = 'Iis:\'
            Filter       = '/system.webServer/security/authentication/windowsAuthentication'
            PropertyName = 'useAppPoolCredentials'
            Value        = $true
            Ensure       = 'Present'
    }
}
```

Pretty much any web configuration property or collection can be managed with those two resources.  Single property collections have proven to be tricky and there is a issue open for that.
